ovirt_imageio/client/_tool.py

```diff
@@ -27,7 +34,23 @@ def main():
         format="%(asctime)s %(levelname)-7s (%(threadName)s) [%(name)s] "
                "%(message)s")
 
-    # XXX Setup signal handlers.
-    # XXX Handle commands errors.
-
-    args.command(args)
+    signal.signal(signal.SIGINT, terminate)
+    signal.signal(signal.SIGTERM, terminate)
+
+    try:
+        args.command(args)
+    except Exception:
+        if terminated:
+            log.error("Terminated by signal %d", terminated)
+            sys.exit(128 + terminated)
+        else:
+            log.exception("Command failed")
+            sys.exit(1)
+
+
+def terminate(signo, frame):
+    global terminated
+    if not terminated:
+        terminated = signo
+        # Terminate qemu-nbd triggering normal cleanup flow.
+        os.killpg(0, signo)
```

Comment:
Is the 0 here a magic shorthand for "my own process group id"? I saw a lot of examples where people explicitly find the id first.

Reply:
0 is not really magic, it means no signals was delivered (since signal number starts at 1) and it is also palsy value.

Reply:
Changing the initial value to None to avoid this confusion.

Reply:
But the first argument to killpg is the process-group-ID.
If I try it in a test program it does kill itself, so I assume it must mean "my group". It just doesn't seem to be very common to use the 0.